### DISCUSSION TITLE
Add dependency into IDE

### DISCUSSION THREAD

**Discussion by jiramares** [VirtusLab/scala-cli#2765] [Scripting]
Hi,
I have file `test.sc` with directive `//> using dep org.scalikejdbc::scalikejdbc:4.2.1`. Is there any way how to add the dependency also into  Intellij Idea other that manually? Command `setup-ide` is not helping.
Thanks 
Jirka

**Comment by adpi2** [comment]
You probably need to reimport the build in IntelliJ by clicking on `Reload All BSP Projects`

![image](https://github.com/VirtusLab/scala-cli/assets/13123162/96619550-e1f9-4ddd-a902-eaad02f11415)

**Reply by jiramares** [reply]
Thanks a lot for your replay but it doesn't seem to help.

If I create empty directory (scala-cli project) and put inside the only file with following content:

    //> using scala 3.3.1
    //> using dep com.ibm.db2:db2jcc4:11.5.7
    //> using dep com.ibm.db2:db2jcc_license_cu:11.5.7
    //> using dep org.scalikejdbc::scalikejdbc:4.2.1
    //> using dep org.slf4j:slf4j-nop:2.0.12
    
    import java.sql.{Connection, DriverManager}
    import scalikejdbc._
    
    Class.forName("com.ibm.db2.jcc.DB2Driver")
    val conn = DriverManager.getConnection("jdbc:db2://tdb:52000/itcards", "db2i11", "*****")
    val db = DB.connect(conn)
    
    val groups: List[String] = db.localTx:
        implicit session =>
            sql"select grp_name from clr.v_group".map(rs => rs.string("GRP_NAME")).list.apply()
    db.close()
    println (groups)

Then I run command `scala-cli setup-ide .` and the `.bsp` directory has been created. Next I open the project in Idea (as exiting one defined by BSP) and unfortunatelly there are no dependecies for `scalikejdbc` defined. I cannot use method name suggestions (`Ctrl + Space`) and the `sql` string interpolation is marked as error.

**Reply by adpi2** [reply]
I tried to reproduce your issue but it worked. (I removed the 2 `db2jcc` libraries that are not available in Maven Central)
 
Are you running the latest versions of Scala CLI, IntelliJ Idea and the IntelliJ Scala Plugin?

**Reply by jiramares** [reply]
> scala-cli --version
    Scala CLI version: 1.1.3
    Scala version (default): 3.3.1


    IntelliJ IDEA 2023.3.4 (Ultimate Edition)
    Build #IU-233.14475.28, built on February 13, 2024


    Scala Plugin 2023.3.27

**Reply by jiramares** [reply]
Can you please tell me whether you have some other versions? My Idea tells everything is up-to-date. Thanks a lot

**Reply by adpi2** [reply]
Hey, I have the exact same set of versions and I am able to import the file without issue.

**Reply by jiramares** [reply]
Thanks a lot for your response. There has to be something strange... 

Before I have reloaded BSP many times and nothing happened. Now I've done reload and the dependencies are there. So it works but don't know why before have not :-(
